src/daemon/rpmostreed-daemon.cxx

```diff
@@ -737,10 +737,21 @@ char *
 rpmostreed_daemon_client_get_sd_unit (RpmostreedDaemon *self, const char *client)
 {
   auto clientdata = static_cast<struct RpmOstreeClient *>(g_hash_table_lookup (self->bus_clients, client));
-  if (!clientdata || clientdata->sd_unit == NULL)
-    return NULL;
-  else
-    return g_strdup (clientdata->sd_unit);
+  /* If the caller didn't register (e.g. Zincati doesn't today because it runs as a
+   * separate UID which systemd doesn't consider `active`), then let's look up its
+   * systemd unit now.
+   */
+  if (!clientdata)
```

Comment:
Minor: Would be slightly nicer to return early in the happy path. Then you can dedent everything else. :)